### PR TITLE
Reflect node.max_local_storage_nodes setting removal

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Follow up to https://github.com/crate/crate/pull/12463
Fixes https://github.com/crate/crate-alerts/issues/249

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
Hi @mfussenegger, could you please take a look? Cannot run nightly locally on macOS due to https://github.com/mfussenegger/cr8/issues/311 - and if I try to use branch:master test runs but timeouts - probably because of local build of distribution. But with `branch:master` actual error `cannot obtain lock` is gone

File: src/crate/qa/tests.py [hunk 1]
```diff
@@ -226,7 +224,6 @@ def upgrade_node(self, old_node, new_version):
         return new_node
 
     def setUp(self):
-        self._path_data = self.mkdtemp()
```
[the remark this answers is not in the document]
Comment (BaurzhanSakhariev) [reply src/crate/qa/tests.py:229]: thx for the hint!

[the remark this answers is not in the document]
Comment (BaurzhanSakhariev) [reply src/crate/qa/tests.py:229]: I now keep this variable, just moved it below to the `new_node`. checked locally 4.8->nightly by replacing nightly to `branch:master`

**Review by mfussenegger — APPROVED** [review]
:+1:

**Pull request merged**